src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -187,6 +187,12 @@ public static object CreateEnumerableValue(ref ReadStack state)
         {
             JsonPropertyInfo jsonPropertyInfo = state.Current.JsonPropertyInfo;
 
+            if (jsonPropertyInfo == null)
+            {
+                Debug.Assert(state.Current.JsonClassInfo.ClassType == ClassType.Object);
+                throw ThrowHelper.GetJsonElementWrongTypeException(JsonTokenType.StartObject, JsonTokenType.StartArray);
```

Comment:
I couldn't figure out how to get the expected `JsonTokenType` from `state.Current`, but in my testing value, array and dictionary types don't throw the NRE. I could only reproduce when I used a class, so I hope this is right.

Reply:
When I added the check in `CreateDictionaryValue`, 191 tests failed.